### DISCUSSION TITLE
[Task]: lightweighting patch core inference

### DISCUSSION THREAD

**Discussion by pure-rgb** [open-edge-platform/anomalib#1323] [Q&A]
### What is the motivation for this task?

In patch core model, in inference time, the heavy backbone is not needed. Thus, if there is any functionality in the inference time for patch-core to remove the backbone, then the inference speed and latency would be greatly improved.

![PatchCore_Architecture](https://github.com/openvinotoolkit/anomalib/assets/45315076/35e69de3-a610-488f-8d73-20ea83f55b2e)

### Describe the solution you'd like

Backbone (resnet) is only used for extracting feature. If it is active in inference time, it will only stay inactive and bright speed and space cost for nothing.  In inference time, it would be effective it is removed or inactive properly.

**Comment by alexriedel1** [comment]
but where would you get your image features at inference time from if the backbone wasn't involved?
The image you posted clearly shows how the image is fed through the "pretrained encoder" at test time.

To increase the inference speed you could use smaller backbones (resnet 18, or even faster and smaller variants like mobilenets)

**Comment by blaz-r** [comment]
As @alexriedel1 pointed out, I don't see how you could get the features needed at test time if you don't have the backbone.

**Comment by samet-akcay** [comment]
@pure-rgb, as @alexriedel1 and @blaz-r pointed out, the backbone is still required to extract the features. Do you want to remove the backbone after extracting the features?

If you refer to the captions of the figure, you could see that anomaly maps are computed from patch-features.
![image](https://github.com/openvinotoolkit/anomalib/assets/17574420/0dafc83d-20d6-4726-a2d2-948d18a46d12)

**Comment by samet-akcay** [comment]
Converting this to discussion, let's continue there.
